src/foam/nanos/auth/UserAndGroupAuthService.js

```diff
@@ -258,9 +248,33 @@ foam.CLASS({
     {
       name: 'validatePassword',
       javaCode: `
-        if ( SafetyUtil.isEmpty(potentialPassword) || ! (Pattern.compile(PASSWORD_VALIDATE_REGEX)).matcher(potentialPassword).matches() ) {
-          throw new RuntimeException(PASSWORD_VALIDATION_ERROR_MESSAGE);
+        PasswordPolicy passwordPolicy = null;
+
+        // Retrieve the password policy from the user and group when available
+        if ( user != null ) {
+          Group ancestor = user.findGroup(x);
+          if ( ancestor != null ) {
```

Comment:
When does this happen? Is it expected than some users won't have a group or is that a system error? If it's the latter I think it could be checked inside findGroup and maybe throw an exception which indicates more criticality than AuthenticationException

Reply:
This would happen if a user's group was deleted, or a user was created with their `group` property set to something that isn't the id of a real group. In both cases though, it's not something we intend to happen.